sentry_sdk/integrations/django/__init__.py

```diff
@@ -314,6 +316,30 @@ def sentry_patched_asgi_handler(self, receive, send):
     AsgiHandler.__call__ = sentry_patched_asgi_handler
 
 
+def _patch_django_asgi_handler():
+    # type: () -> None
+    try:
+        from django.core.handlers.asgi import ASGIHandler
+    except ImportError:
+        return
+
+    if not HAS_REAL_CONTEXTVARS:
+        # We better have contextvars or we're going to leak state between
+        # requests.
+        #
+        # We cannot hard-raise here because Django may not be used at all in
+        # the current process.
+        logger.warning(
+            "We detected that you are using Django 3. To get proper "
+            "instrumentation for ASGI requests, the Sentry SDK requires "
+            "Python 3.7+ or the aiocontextvars package from PyPI."
```

Comment:
Is it enough to `import aiocontextvars` for side-effects in https://github.com/getsentry/sentry-python/blob/85c15e2e6d9707e43a412f0b49a530aab8925624/sentry_sdk/utils.py#L741-L759?

I'm new to that, please bear with me :)

Reply:
`aiocontextvars` is a third-party package that offers a stand-in replacement for Python 3.7's `contextvars` in the stdlib. It is generally preferred over `threading.local` because it actually works with asyncio, *however*, in applications that use gunicorn/gevent/eventlet, `threading.local` works better than either contextvars or aiocontextvars. The reason for *that* is because gevent monkeypatches the threading module to behave correctly, but is not doing the same thing for contextvars or aiocontextvars.

This is why this logic became that contrived. We test this by having multiple virtualenvs with gevent/eventlet installed/not installed and initialized/not initialized.

Reply:
note: check if aiocontextvars actually works

Reply:
It did work, but is unreadable. Follow up is https://github.com/getsentry/sentry-python/pull/574